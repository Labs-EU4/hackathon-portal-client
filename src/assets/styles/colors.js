//COLORS
export const white = '#F9F9F9';
export const black = '#010038';

//RED COLOR
export const red = '#E34449';
export const brightRed = '#F34D21';

//YELLOW COLOR
export const dimYellow = '#ffdb00';

//GREY COLOR
export const grey = '#39444D';
export const textLightGrey = '#999';
export const textDarkGrey = '#666';
export const bgLightGrey = '#f0f0f0';
export const bgGrey = '#dfe3e6';
export const borderGrey = '#39444D';
// export const lightGrey = '#F5F5F5';

//GREEN COLOR
export const lightGreen = '#afeeee';
export const green = '#5aac44';
export const brightgreen = '#25d366';
export const darkGreen = '#1b9849';


// export const primary = '#04c8de'; //--> same as the above, good combo
// export const primary = '#F5F5F5'; //--> same as the above, good combo
// export const primary = '#FF3636'; //--> same as the above, good combo
// export const primary = '#47e4bb'; //--> same as the above, good combo

//PRIMARY COLOR
export const primaryLight = '#81DBFD';
export const primary = '#03AEEF';
export const primaryDark = '#03AEEF';
// export const primary = '#8CCAE9';
// export const primaryLight = '#04c8de';
// export const primary = '#40e0d0';

//SECONDARY COLOR
export const secondaryLight = '#F8C485';
export const secondary = '#f39422';
export const secondaryDark = '#DA7C0A';
// export const secondaryDark = '#F34D21';






export const backgroundBlack = '#222';
export const backgroundWhite = '#f4f4f4';
export const lightBlue = '#0ff';

// export const white = '#f4f7f6'; // --> GOOD WHITE
export const textWhite = '#F8F9FA';
export const primaryWhite = '#eff';
export const secondaryWhite = '#eee';
export const tertiaryWhite = '#ddd';
export const quaternaryWhite = '#ccc';
export const senaryWhite = '#aaa';




// export const black = '#000';
// export const black = '#1F1C05'; --> VERY GOOD BLACK SHADE
// export const black = '#020404 '; --> ALRIGHT BLACK!!!
// export const black = '#252525'; //--> SUPER GOOD BLACK
// export const black = '#202121';
export const softGreen = '#aec';
export const spaceBlue= '#03AEEF';
export const brightPurple= '#4444dd';
export const darkTwitterTheme = '#15202B';
export const lightDarkTwitterInputBg = '253341#';
export const fbCopyCatLightTheme = '#F3F4F6';
export const youTubeColorBlue = '#065FD4';
export const flickrBlue = '#1396E7';
export const btnBlue = '#0973FF';

//Add theme using styled components

//MAIN THEME COLORS
// export const primaryColor = '#BDE6EC';
// export const primaryColor = '#aec';
// export const primaryColor = '#ffdb00';
// export const primaryColor = '#6747C7'; //--> SUPER GOOD!!
// export const primaryColor = '#f8f4ff'; //--> GOOD WHITE VERSION!!
// export const primaryColor = '#fada5e'; //--> ROYAL YELLOW
// export const lightPrimaryColor = '#FEF33E'; //--> BRIGHT COOL YELLOW
// export const primaryColor = '#FFFF01'; //--> VERY BRIGHT YELLOW
// export const primaryColor = '#F4B75D'; //--> VERY BRIGHT YELLOW
// export const primaryColor = '#448ef6'; //--> VERY BRIGHT YELLOW
// export const primaryColor = '#6decb9'; //--> VERY BRIGHT YELLOW


// export const primaryColor = '#537ec5'; //--> super good color, mostly used theme!!!!
// export const turquoise = '#ff5edf'; //--> change to secondary color

export const tertiaryColor = '#5B5F97';
export const tertiaryColorDarker = '#484b77';

export const borderGrayTheme = '#39444D';

// export const primaryColor = '#1396E7'; //--> VERY GOOOD
// export const primaryColor = '#065FD4'; //--> VERY good blue, 2nd/3rd
// export const primaryColor = '#FFf'; //--> MOST USED YELLOW!!!! 
// export const primaryColor = '#B390EE';
// export const primaryColor = '#CC2D8A';
// export const primaryColor = '#46130B';
// export const primaryColor = '#FAD52D';

// export const turquoise = '#40e0d0 ';
// export const turquoise = '#4dd599'; --> VERY GOOD
// export const turquoise = '#47e4bb'; --> EVEN BETTER
// export const turquoise = '#3fc5f0';
//DARK-MODE TOGGLE COLORS
export const sunrise = '#ffc922';
export const yellowSun = '#ffdb00';
export const blueMoon = '#003366';
export const papayawhip = 'papayawhip';


// COOL COLORS 
export const newSecondaryColor = '#B390EE';
export const newTertiaryColor = '#B0EEEE';


// DARK MODE
export const MainBackgroundColor = '#29272F';
