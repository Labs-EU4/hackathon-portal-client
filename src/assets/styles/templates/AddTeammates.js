import styled from "styled-components";

import { media } from '../variables/media';
import BodyContainer from "../atoms/BodyContainer";
import { CardWide } from "../atoms/Card";
import Icon from '../../../components/atoms/Icon';

export const BodyContainerColumn = styled(BodyContainer)`
  flex-direction: column;
  justify-content: start;
`;

export const StyledContainer = styled.div`
  display: block;
  position: relative;
`;

export const StyledWidget = styled.div`
  margin-bottom: 10px;
  cursor: pointer;
`;

export const ContainerRadio = styled.label`
  /* Customize the label (the container) */
  display: block;
  position: relative;
  padding-left: 35px;
  margin-bottom: 12px;
  cursor: pointer;
  font-size: 22px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;

  /* Hide the browser's default radio button */
  input {
    position: absolute;
    opacity: 0;
    cursor: pointer;
    height: 0;
    width: 0;

    /* When the radio button is checked, add a blue background */
    &:checked ~ span {
      background-color: ${props => props.theme.color.blue.bg};
    }

    /* Show the indicator (dot/circle) when checked */
    &:checked ~ span:after {
      display: block;
    }
  }

  /* Create a custom radio button */
  span {
    position: absolute;
    top: 0;
    left: 0;
    height: 25px;
    width: 25px;
    background-color: #eee;
    border-radius: 50%;

    /* Create the indicator (the dot/circle - hidden when not checked) */
    &:after {
      content: "";
      position: absolute;
      display: none;

      /* Style the indicator (dot/circle) */
      top: 9px;
      left: 9px;
      width: 8px;
      height: 8px;
      border-radius: 50%;
      background: white;
    }
  }

  /* On mouse-over, add a grey background color */
  &:hover {
    input ~ span {
      background-color: #ccc;
    }
  }
`;


//SEARCH WIDGET
export const Container = styled.div`
  position: relative;

  input {
    font-size: 1.6rem;
    font-weight: 500;
    color: ${props => props.theme.color.black.regular};
    border: 2px solid ${props => props.theme.color.grey.regular};
    border-radius: 6px;
    padding: 10px;
    ${({ display }) =>
    display === "wide" ? `width: 100%;` : `width: 180px;`}

    &:focus {
      border: 2px solid ${props => props.theme.color.primary.regular};
      transition: all 0.4s ease;
      box-shadow: 0 0 3px #ddd;

      & + svg {
        color: ${props => props.theme.color.primary.regular};
      }
    }
  }

  @media ${media.tablet} {
    width: 100%;
    margin-right: 0;
  }
`;

export const UsersList = styled.div`
  ${props => props.theme.shadow.box};
  width: 100%; height: 48.25vh;
  border: 2px solid ${props => props.theme.color.primary.regular}; border-bottom: none;
  margin: 10px 0 0;
  padding-bottom: 15px;
  overflow-y: scroll;
  user-select: none;

  &::-webkit-scrollbar-button {
    background-color: blue;
  }
`;

export const ChosenJudgesContainer = styled.div`
  position: relative;
  display: flex; align-items: center;
  width: 100%; height: 45px;
  background-color: ${props => props.theme.color.white.regular};
  border: 2px solid ${props => props.theme.color.primary.regular};
  border-radius: 20px;
  margin-top: -15px; margin-bottom: 20px;
  box-shadow: 1px 1px 1px rgba(0, 0, 0, .4);
  overflow-x: scroll;
  z-index: 500;
`;

export const ChosenJudgeImg = styled.img`
  width: 33px; height: 33px;
  border-radius: 50%;
  margin-left: 5px;
  object-fit: cover;
`;

export const StyledSearchIcon = styled(Icon)`
  position: absolute; top: 10px; left: calc(100% - 30px);
  transform: rotate(90deg);
  cursor: pointer;

  &:hover {
    transform: scale(1.1) rotate(90deg);
  }
`;

export const StyledWideBody = styled.div`
  position: absolute; top: 0; left: 0;
  width: 100%; height: 100%;
  background-color: rgba(0, 0, 0, .4);
  z-index: 100;
`;

export const StyledCardWide = styled(CardWide)`
  ${props => props.theme.shadow.box};
  position: absolute; top: 50%; left: 50%;
  width: 500px; 
  transform: translate(-50%, -50%);
  z-index: 200;
`;

//JUDGE WIDGET
export const StyledJudgeWidget = styled.div`
  position: sticky; top: 0; left: 0;
  display: flex;
  width: calc(100% - 10px);
  background-color: ${props => props.theme.color.white.regular};
  border: 2px solid ${props => props.theme.color.black.regular};
  border-radius: 3px;
  padding: 5px 10px;
  margin-bottom: 2px;
  cursor: pointer;

  &:hover {
    background-color:  ${props => props.theme.color.primary.regular};
    color: ${props => props.theme.color.white.regular};
  }

  &.selected {
    background-color: ${props => props.theme.color.primary.regular};
    opacity: .5;
  }
`;

export const UserAvatar = styled.figure`
  width: 50px; height: 50px;
  border-radius: 50%;
  object-fit: cover;
  margin-right: 10px;

  & > img {
    width: 100%;
  }
`;

export const UserInfo = styled.div`
  ${props => props.theme.flex.custom('center', 'flex-start', 'column')};
  margin-left: 5px;

  p {
    &:last-child {
      ${props => props.theme.fontSize.small};
      font-weight: normal;
    }
  }
`;