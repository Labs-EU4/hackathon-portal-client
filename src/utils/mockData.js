export const initialState = {
  events: {
    data: [
      {
        id: 1,
        event_title: "EuroHack",
        event_description: "the best event hack hack hack macbook",
        creator_id: 1,
        start_date: "10/09/11",
        end_date: "08/07/13",
        location: "Europe",
        guidelines: "cool hekhkdsjfhsdkjf skdjfhskdjfhkjsdfh skjdfhksjdfh",
        participation_type: "Team",
        category_id: 1,
        requirements: [
          {
            video_url:
              "https://res.cloudinary.com/ibkrice-com/video/upload/v1581259780/Harvesting_ifato8.mp4"
          }
        ],
        ProfileImage: [
          {
            image_url:
              "https://res.cloudinary.com/ibkrice-com/video/upload/v1581259780/Harvesting_ifato8.mp4"
          }
        ],
        rubrics: [
          "Presentation",
          "Product Fit",
          "Functionality",
          "Innovation",
          "Product Design",
          "Extensibility"
        ]
      },
      {
        id: 2,
        event_title: "World",
        event_description: "number one hacker event in the world",
        creator_id: 1,
        start_date: "07/09/11",
        end_date: "08/07/13",
        location: "Asia",
        guidelines: "come well prepared,no cheating,no fighting",
        participation_type: "Team",
        category_id: 2
      }
    ],
    categories: [
      { id: 2, category_name: "Summer Hackton" },
      { id: 3, category_name: "World Hackton" },
      { id: 3, category_name: "Asia Hackton" },
      { id: 3, category_name: "Gamer's Hackton" },
      { id: 3, category_name: "Advanced Hackton" }
    ]
  },

  currentUser: {
    token: 347534574536,
    userId: 1,
    fullname: "Jake Tom",
    email: "8omemail@google.com",
    username: "Jake22",
    bio: "I like coding ",
    image: ""
  },
  submissions: [
    {
      id: 1,
      project_title: "Mock Project",
      participant_or_team_name: "Mock Team",
      event_id: 7,
      video_url: "",
      git_url: "https://www.google.com/",
      project_writeups: "No mock write up",
      submitted_by: 14,
      created_at: "2020-03-02",
      updated_at: "2020-03-02",
      average_rating: 0,
      acted_judges: 0,
      number_of_judges: 0
    }
  ],

  teams: [
    { id: 1, team_name: "The A team", team_lead: 2 },
    { id: 2, team_name: "Hacking central", team_lead: 3 },
    { id: 3, team_name: "blackhats", team_lead: 4 },
    { id: 4, team_name: "Spies", team_lead: 1 }
  ],
  participant_team_members: [
    {
      team_member: 3,
      team_id: 1
    },
    {
      team_member: 4,
      team_id: 1
    },
    {
      team_member: 5,
      team_id: 1
    }
  ]
};
