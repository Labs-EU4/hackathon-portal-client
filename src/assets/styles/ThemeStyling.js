import * as color from './variables/colors';

const size = {
    mobileS: '320px',
    mobileM: '375px',
    mobileL: '425px',
    tablet: '768px',
    laptop: '1024px',
    laptopL: '1440px',
    desktop: '2560px'
};

export const theme = {
    color: {
        primary: {
            light: '#81DBFD',
            regular: color.primary,
            // dark: color.primaryDark
        },
        secondary: {
            // light: color.secondaryLight,
            regular: color.secondary,
            // dark: color.secondaryDark
        },
        white: {
            regular: color.white,
            bg: color.bgWhite
        },
        black: {
            light: '#212121',
            regular: color.black,
        },
        red: {
            regular: color.red,
            bright: color.brightRed
        },
        blue: {
            light: color.lightBlue,
            regular: color.blue,
            bg: '#2196f3'
        },
        grey: {
            light: '#696969',
            regular: color.grey,
            dark: '#444',
            bg: color.bgGrey,
            border: 'rgba(14,19,24,.15)'
        },
        green: {
            light: color.lightGreen,
            regular: color.green,
            dark: color.darkGreen
        },
        success: color.success,
        warning: color.warning,
        danger: color.danger,
        info: color.info,
        link: {
            hover: 'rgba(14,19,24, .07)',
        }
    },
    fontSize: {
        smaller: "font-size: 1.2rem",
        small: "font-size: 1.4rem;",
        base: "font-size: 1.6rem;",
        medium: "font-size: 1.8rem",
        large: "font-size: 2rem",
        h1: "font-size: 7.5rem;",
        h2: "font-size: 5.5rem;",
        h3: "font-size: 3.5rem;",
        h4: "font-size: 2.5rem;",
        button: "font-size: 1.8rem;"
    },
    shadow: {
        text: `text-shadow: .5px .5px 1px ${color.black}`,
        text_links: `1px .75px 1.5px ${color.black}`,
        box: `box-shadow: 0 3px 20px 0 rgba(0, 0, 0, .2)`,
        card: `box-shadow: 1px .5px 3px rgba(0, 0, 0, .15)`,
        intenseBox: `box-shadow: 0 2.5px 10px ${color.black};`,
        filter: `filter: drop-shadow(.15px .15px 1.5px ${color.black});
        filter: drop-shadow(-.15px -.15px 1px ${color.black});`,
        btn: `2px 2px 3px ${color.black}`
    },
    flex: {
        center: 'display: flex; justify-content: center; align-items: center;',
        columnCenter: 'display: flex; flex-direction: column; justify-content: center; align-items: center;',
        column: 'display: flex; flex-direction: column;',

        custom: (justify='start', align='start', direction='row', wrap='no-wrap') => {
            return `
                display: flex; 
                flex-direction: ${direction}
                justify-content: ${justify}; 
                align-items: ${align};
                flex-wrap: ${wrap};
            `
        },
    },
    device: {
        mobileS: `(min-width: ${size.mobileS})`,
        mobileM: `(min-width: ${size.mobileM})`,
        mobileL: `(min-width: ${size.mobileL})`,
        tablet: `(min-width: ${size.tablet})`,
        laptop: `(min-width: ${size.laptop})`,
        laptopL: `(min-width: ${size.laptopL})`,
        desktop: `(min-width: ${size.desktop})`,
        desktopL: `(min-width: ${size.desktop})`
    }
};