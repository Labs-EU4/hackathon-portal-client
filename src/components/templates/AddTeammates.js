import React, { useState, useRef } from "react";
import { useDispatch } from "react-redux";
import { useParams, useHistory } from "react-router-dom";

import { StyledWideBody, StyledCardWide } from '../../assets/styles/templates/AddTeammatesStyling';
import { H3 } from "../../assets/styles/atoms/HeadingStyling";
import { RowHead } from "../../assets/styles/atoms/RowHeadStyling";
import { Column } from "../../assets/styles/atoms/ColumnStyling";
import { addTeamMember, sendEventTeamInvite } from "../../store/events/actions";
import { JudgesSearchWidget, TeamRoleWidget, TeamInviteWidget } from "./widgets";

const AddTeammates = (
  // { id, setEventId, setIsEventModalOpen, setIsAddJudgeOpen }
) => {
  const selectedUserArr = useRef([]);
  const role = "judge";
  const dispatch = useDispatch();
  const [noneUser, setNoneUser] = useState(null);
  const history = useHistory();
  const { id } = useParams();

  const selectedUsersHandler = async(addedUser) => {
    const newArray = [...selectedUserArr.current, addedUser];
    selectedUserArr.current = newArray;
  }

  const handleSubmit = () => {
    selectedUserArr.current.map(selectedUser => {
      const { email } = selectedUser;
      const data = {
        eventId: Number(id),
        email,
        role
      };
      return dispatch(addTeamMember(data, history));
    });
    // setIsEventModalOpen(false);
    // setIsAddJudgeOpen(false);
    // setEventId(null);
  };

  const handleExit = () => {
    //!!See if it works, otherwise make changes
    history.back();
    // setIsEventModalOpen(true);
    // setIsAddJudgeOpen(false);
  };

  const sendInvite = () => {
    const data = {
      eventId: Number(id),
      email: noneUser,
      role
    };
    dispatch(sendEventTeamInvite(data, history));
  };

  return (
    <StyledWideBody>
      <Column>
        <StyledCardWide>
          <RowHead>
            <H3>Add Judge</H3>
          </RowHead>
          <JudgesSearchWidget 
            {...{selectedUsersHandler}}
            {...{selectedUserArr}}
            {...{setNoneUser}}
            {...{handleExit}}
            {...{handleSubmit}}
          />
          {noneUser && (
            <TeamInviteWidget 
              {...{noneUser}}
              {...{sendInvite}}
            />
          )}
        </StyledCardWide>
      </Column>
    </StyledWideBody>
  );
};

export default AddTeammates;


// import React, { useState } from "react";
// import { useDispatch } from "react-redux";
// import { useParams } from "react-router-dom";
// import { Footer } from "../organisms/index";
// import UserHeader from "../organisms/UserHeader";
// import WideBody from "../atoms/WideBody";
// import { H3 } from "../atoms/Heading";
// import { RowHead } from "../atoms/RowHead";
// import { Column } from "../atoms/Column";
// import { CardWide } from "../atoms/Card";
// import { addTeamMember, sendEventTeamInvite } from "../../store/events/actions";
// import Nav from "../molecules/Nav";
// import { BodyContainerColumn } from "../styles/templates/AddTeammatesStyling";
// import { SearchWidget, TeamRoleWidget, TeamInviteWidget } from "./widgets";
// import { useHistory } from "react-router-dom";

// const AddTeammates = () => {
//   const [selectedUser, setSelectedUser] = useState(null);
//   const [role, setRole] = useState("");
//   const dispatch = useDispatch();
//   const { id } = useParams();
//   const [noneUser, setNoneUser] = useState(null);
//   const history = useHistory();

//   const handleSubmit = () => {
//     const { email } = selectedUser;
//     const data = {
//       eventId: Number(id),
//       email,
//       role
//     };
//     dispatch(addTeamMember(data, history));
//   };

//   const sendInvite = () => {
//     const data = {
//       eventId: Number(id),
//       email: noneUser,
//       role
//     };
//     dispatch(sendEventTeamInvite(data, history));
//   };

//   return (
//     <div>
//       <UserHeader />
//       <WideBody>
//         <Nav />
//         <BodyContainerColumn>
//           <RowHead>
//             <H3>Add Teammates</H3>
//           </RowHead>
//           <Column>
//             <CardWide>
//               {
//                 !selectedUser ? (
//                   <SearchWidget
//                     setSelectedUser={setSelectedUser}
//                     setNoneUser={setNoneUser}
//                   />
//                 ) : (
//                     <TeamRoleWidget
//                       setRole={setRole}
//                       role={role}
//                       handleSubmit={handleSubmit}
//                     />
//                   )

//               }
//               {noneUser ? (
//                 <TeamInviteWidget noneUser={noneUser} sendInvite={sendInvite} />
//               ) : null}
//             </CardWide>
//           </Column>
//         </BodyContainerColumn>
//       </WideBody>
//       <Footer />
//     </div>
//   );
// };

// export default AddTeammates;
