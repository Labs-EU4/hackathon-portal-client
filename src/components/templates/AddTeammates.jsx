import React, { useState, useEffect } from "react";
import isEmail from "validator/lib/isEmail";
import { useDispatch } from "react-redux";
import { useParams } from "react-router-dom";
import { Footer } from "../organisms/index";
import UserHeader from "../organisms/UserHeader";
import WideBody from "../atoms/WideBody";
import { H3 } from "../atoms/Heading";
import { RowHead } from "../atoms/RowHead";
import { Column } from "../atoms/Column";
import { CardWide } from "../atoms/Card";
import { addTeamMember, sendEventTeamInvite } from "../../store/events/actions";
import Nav from "../molecules/Nav";
import { BodyContainerColumn } from "../styles/templates/AddTeammatesStyling";
import { SearchWidget, RoleWidget, InviteWidget } from "./widgets";

const AddTeammates = () => {
  const [selectedUser, setSelectedUser] = useState(null);
  const [role] = useState("");
  const dispatch = useDispatch();
  const { id } = useParams();
  const [noneUser, setNoneUser] = useState(null);

  const handleSubmit = () => {
    const { email } = selectedUser;
    const data = {
      eventId: Number(id),
      email,
      role
    };
    dispatch(addTeamMember(data, history));
  };

  const sendInvite = () => {
    const data = {
      eventId: Number(id),
      email: noneUser,
      role
    };
    dispatch(sendEventTeamInvite(data, history));
  };

  return (
    <div>
      <UserHeader />
      <WideBody>
        <Nav />
        <BodyContainerColumn>
          <RowHead>
            <H3>Add Teammates</H3>
          </RowHead>
          <Column>
            <CardWide>
              {
                !selectedUser ? (
                  <SearchWidget
                    setSelectedUser={setSelectedUser}
                    setNoneUser={setNoneUser}
                  />
                ) : (
                    <RoleWidget
                      selectedUser={selectedUser}
                      handleSubmit={handleSubmit}
                    />
                  )

              }
              {noneUser ? (
                <InviteWidget noneUser={noneUser} sendInvite={sendInvite} />
              ) : null}
            </CardWide>
          </Column>
        </BodyContainerColumn>
      </WideBody>
      <Footer />
    </div>
  );
};

export default AddTeammates;
