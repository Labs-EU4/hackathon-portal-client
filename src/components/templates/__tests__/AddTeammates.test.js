import React from "react";
import { Router } from "react-router";
import { createMemoryHistory } from "history";
import { Provider } from "react-redux";
import { render, fireEvent, cleanup } from "@testing-library/react";
import configureStore from "redux-mock-store";
import "@testing-library/jest-dom/extend-expect";
import AddTeammates from "../AddTeammates";
import { initialState } from "../../../utils/mockData";
import { theme } from "../../../assets/styles/ThemeStyling";
import { ThemeProvider } from "styled-components";
import { library } from '@fortawesome/fontawesome-svg-core';
import { fas } from '@fortawesome/free-solid-svg-icons';
import { far } from '@fortawesome/free-regular-svg-icons';
import { fab } from '@fortawesome/free-brands-svg-icons';

library.add(fas, far, fab);


const history = createMemoryHistory();

afterEach(cleanup);

let jestFeatures;
let mockStore;
let store;

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"), // use actual for all non-hook parts
  useParams: () => ({
    id: 1
  })
}));

beforeEach(() => {
  mockStore = configureStore();
  store = mockStore(initialState);
  jestFeatures = render(
    <Router history={history}>
      <Provider store={store}>
        <ThemeProvider theme={theme}>
          <AddTeammates />
        </ThemeProvider>
      </Provider>
    </Router>
  );
});

describe("Shows all the text nodes on AddTeammates.js that are contained on the making an event form", () => {
  it("should be displaying the 'Add Judge' text node", () => {
    expect(jestFeatures.getByTestId("heading")).toBeInTheDocument();
  });
  it("should be displaying the 'Back' text node", () => {
    expect(jestFeatures.getByText(/Back/i)).toBeInTheDocument();
  });
  it("asserts that the component renders properly", () => {
    expect(jestFeatures).toMatchSnapshot();
  });
});
