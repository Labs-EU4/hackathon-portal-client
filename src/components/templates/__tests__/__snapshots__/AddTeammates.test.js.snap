// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shows all the text nodes on AddTeammates.js that are contained on the making an event form asserts that the component renders properly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="sc-iRbamj liaLaB"
        >
          <div
            class="sc-gipzik gYIzbI"
          >
            <a
              href="/dashboard"
            >
              <figure
                class="sc-bRBYWo czjYxr"
              />
            </a>
            <div
              class="sc-hXRMBi Vyllq"
            >
              <div
                class="sc-kjoXOD LrRDY"
              >
                <span>
                  <svg>
                    link-burger-icon.svg
                  </svg>
                  Menu
                </span>
                <ul>
                  <li>
                    <a
                      class="sc-cHGsZl bMwAHR"
                      href="/dashboard"
                    >
                      Dashboard
                    </a>
                  </li>
                  <li>
                    <a
                      class="sc-cHGsZl bMwAHR"
                      href="/dashboard/profile"
                    >
                      Profile
                    </a>
                  </li>
                </ul>
              </div>
              <img
                class="sc-hqyNC gZXmwY"
                src="user_icon.svg"
              />
            </div>
          </div>
        </div>
        <div
          class="sc-fMiknA gLARYc"
        >
          <div
            class="sc-gisBJw dVDYcw"
          >
            <a
              class="sc-cHGsZl bMwAHR"
              href="/dashboard"
            >
              <svg>
                link-dashboard-icon.svg
              </svg>
              Dashboard
            </a>
            <a
              class="sc-cHGsZl bMwAHR"
              href="/dashboard/profile"
            >
              <svg>
                link-profile-icon.svg
              </svg>
              Profile
            </a>
          </div>
          <div
            class="sc-eqIVtm hvmLNT sc-cIShpX dTlsAK"
          >
            <div
              class="sc-dNLxif imKLHn"
            >
              <h3
                class="sc-kGXeez lolzGQ"
              >
                Add Teammates
              </h3>
            </div>
            <div
              class="sc-uJMKN eujmAt"
            >
              <div
                class="sc-gGBfsJ sc-jnlKLf kvMLTu"
              >
                <div
                  class="sc-eXEjpC zsXnf"
                  display="wide"
                >
                  <input
                    placeholder="Search by email"
                    type="text"
                    value=""
                  />
                  <button
                    class="sc-kEYyzF kbIspq"
                    color="grey"
                  >
                    Back to dashboard
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="sc-bdVaJa bDiKvh"
        >
          <div
            class="sc-htpNat fAaFDY"
          >
            <div>
              <img
                alt="LinkedIn"
                class="sc-ifAKCX zCzIi"
                src="Icon-linkedin.png"
              />
              <img
                alt="Twitter"
                class="sc-ifAKCX zCzIi"
                src="Icon-twitter.png"
              />
              <img
                alt="Facebook"
                class="sc-ifAKCX zCzIi"
                src="Icon-facebook.png"
              />
            </div>
            <div>
              <a
                class="sc-gzVnrw sc-dnqmqq bGScqw"
                href="https://hackton.co/#team"
              >
                Team
              </a>
              <a
                class="sc-gzVnrw sc-dnqmqq bGScqw"
                href="#"
              >
                Privacy
              </a>
              <a
                class="sc-gzVnrw sc-dnqmqq bGScqw"
                href="#"
              >
                Contacts
              </a>
            </div>
            <div>
              <p
                class="sc-VigVT sc-jTzLTM gBKmld"
              >
                International Crafters © 2020
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="sc-iRbamj liaLaB"
      >
        <div
          class="sc-gipzik gYIzbI"
        >
          <a
            href="/dashboard"
          >
            <figure
              class="sc-bRBYWo czjYxr"
            />
          </a>
          <div
            class="sc-hXRMBi Vyllq"
          >
            <div
              class="sc-kjoXOD LrRDY"
            >
              <span>
                <svg>
                  link-burger-icon.svg
                </svg>
                Menu
              </span>
              <ul>
                <li>
                  <a
                    class="sc-cHGsZl bMwAHR"
                    href="/dashboard"
                  >
                    Dashboard
                  </a>
                </li>
                <li>
                  <a
                    class="sc-cHGsZl bMwAHR"
                    href="/dashboard/profile"
                  >
                    Profile
                  </a>
                </li>
              </ul>
            </div>
            <img
              class="sc-hqyNC gZXmwY"
              src="user_icon.svg"
            />
          </div>
        </div>
      </div>
      <div
        class="sc-fMiknA gLARYc"
      >
        <div
          class="sc-gisBJw dVDYcw"
        >
          <a
            class="sc-cHGsZl bMwAHR"
            href="/dashboard"
          >
            <svg>
              link-dashboard-icon.svg
            </svg>
            Dashboard
          </a>
          <a
            class="sc-cHGsZl bMwAHR"
            href="/dashboard/profile"
          >
            <svg>
              link-profile-icon.svg
            </svg>
            Profile
          </a>
        </div>
        <div
          class="sc-eqIVtm hvmLNT sc-cIShpX dTlsAK"
        >
          <div
            class="sc-dNLxif imKLHn"
          >
            <h3
              class="sc-kGXeez lolzGQ"
            >
              Add Teammates
            </h3>
          </div>
          <div
            class="sc-uJMKN eujmAt"
          >
            <div
              class="sc-gGBfsJ sc-jnlKLf kvMLTu"
            >
              <div
                class="sc-eXEjpC zsXnf"
                display="wide"
              >
                <input
                  placeholder="Search by email"
                  type="text"
                  value=""
                />
                <button
                  class="sc-kEYyzF kbIspq"
                  color="grey"
                >
                  Back to dashboard
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="sc-bdVaJa bDiKvh"
      >
        <div
          class="sc-htpNat fAaFDY"
        >
          <div>
            <img
              alt="LinkedIn"
              class="sc-ifAKCX zCzIi"
              src="Icon-linkedin.png"
            />
            <img
              alt="Twitter"
              class="sc-ifAKCX zCzIi"
              src="Icon-twitter.png"
            />
            <img
              alt="Facebook"
              class="sc-ifAKCX zCzIi"
              src="Icon-facebook.png"
            />
          </div>
          <div>
            <a
              class="sc-gzVnrw sc-dnqmqq bGScqw"
              href="https://hackton.co/#team"
            >
              Team
            </a>
            <a
              class="sc-gzVnrw sc-dnqmqq bGScqw"
              href="#"
            >
              Privacy
            </a>
            <a
              class="sc-gzVnrw sc-dnqmqq bGScqw"
              href="#"
            >
              Contacts
            </a>
          </div>
          <div>
            <p
              class="sc-VigVT sc-jTzLTM gBKmld"
            >
              International Crafters © 2020
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
